### ISSUE TITLE
Rancher Server HA on Galera - "This table type requires a primary key"

### ISSUE THREAD

**Issue by gifkinsj** [rancher/rancher#8087]
**Rancher Versions:**
Server: v1.4.1

**Docker Version:**
Docker version 1.10.3

**OS and where are the hosts located? (cloud, bare metal, etc):**
CentOS 7 - VMware

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
HA Rancher, external DB (MariaDB Galera, multi-master)

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Configure Rancher HA using a MariaDB Galera cluster as the database back end. When attempting to configure the database Rancher crashes due to Galera requiring primary keys on InnoDB tables. (This is caused by innodb_force_primary_key=1, which is required for a Galera cluster)

**Results:** 
Rancher Server crashes and the database isn't provisioned.

**Comment by gifkinsj** [comment]
I've created the database on a non-Galera node and dumped the schema to file. Upon attempting to import the file to the Galera cluster, I found I had to add a primary key to the "DATABASECHANGELOG" table.

After doing so the script was able to execute, however I ran into further errors  "Foreign key constraint is incorrectly formed" per attached.

(Note my database is named "rancher", not the default).

[rancher-sql.zip](https://github.com/rancher/rancher/files/826273/rancher-sql.zip)
